instrumentation/resque/lib/opentelemetry/instrumentation/resque/patches/resque_job.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Resque
+      module Patches
+        # Module to prepend to Resque::Job for instrumentation
+        module ResqueJob
+          def perform # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+            job_class = payload_class_name
+
+            attributes = {
+              'messaging.system' => 'resque',
+              'messaging.destination' => queue.to_s,
+              'messaging.destination_kind' => 'queue',
+              'messaging.resque.job_class' => job_class
+            }
+
+            span_name = if config[:job_class_span_names]
+                          "#{job_class} process"
+                        else
+                          "#{queue} process"
+                        end
+
+            extracted_context = OpenTelemetry.propagation.extract(@payload)
+            span_context = OpenTelemetry::Trace.current_span(extracted_context).context
+
+            links = []
+            links << OpenTelemetry::Trace::Link.new(span_context) if span_context.valid?
+
+            if config[:propagate_context]
+              OpenTelemetry::Context.with_current(extracted_context) do
+                tracer.in_span(span_name, attributes: attributes, links: links, kind: :consumer) do
+                  super
+                end
+              end
+            else
+              tracer.in_span(span_name, attributes: attributes, links: links, kind: :consumer) do
```

Comment:
I think we should only have the link if we're starting a new trace. If we're continuing the trace, the link is implicitly provided by the parent context.

```suggestion
            if config[:propagate_context]
              OpenTelemetry::Context.with_current(extracted_context) do
                tracer.in_span(span_name, attributes: attributes, kind: :consumer) do
                  super
                end
              end
            else
              links = []
              links << OpenTelemetry::Trace::Link.new(span_context) if span_context.valid?
              tracer.in_span(span_name, attributes: attributes, links: links, kind: :consumer) do
```